config/opa-service/v3.yaml

```diff
@@ -31,9 +31,141 @@ static_resources:
                   cluster_name: ext_authz-opa-service
                 timeout: 0.250s
               transport_api_version: V3
+          - name: envoy.filters.http.lua
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+              inline_code: |
+                function envoy_on_request(request_handle)
+                  decodejwt = (loadfile "/etc/envoy-config/opa-service/lua/decodejwt.lua")()
+                  JSON = (loadfile "/etc/envoy-config/opa-service/lua/JSON.lua")()
+                  chunks = {}
+                  raw_auth_header = request_handle:headers():get("authorization")
+                  for substring in raw_auth_header:gmatch("%S+") do
+                     table.insert(chunks, substring)
+                  end
+                  token = decodejwt:decode_jwt(chunks[2])
+                  content_type = request_handle:headers():get("content-type")
+                  if content_type == "application/json" then
+                    local body = request_handle:body()
+                    local body_size = body:length()
+                    local body_bytes = body:getBytes(0, body_size)
+                    local raw_json_text = tostring(body_bytes)
+                    lua_value = JSON:decode(raw_json_text)
+                    access_to = lua_value.id
```

Comment:
.id is valid for payloads like NGSI-LD, but may not be a good for other types.

how can we make this script easy to configure?

Reply:
k i got that this is totally api dependent also because it's actually chained to an api

Reply:
Yes, these are per configured "route", so they won't interfere with one another. I'll try to move some code in a common lua file at least

Reply:
should this be covered in a new pr?